sharedUiComponents/lines/color3LineComponent.tsx

```diff
@@ -17,47 +17,99 @@ export interface IColor3LineComponentProps {
     propertyName: string;
     onPropertyChangedObservable?: Observable<PropertyChangedEvent>;
     isLinear?: boolean;
-    icon? : string;
+    icon?: string;
     lockObject?: LockObject;
-    iconLabel? : string;
+    iconLabel?: string;
+    onValueChange?: (value: string) => void;
 }
 
-export class Color3LineComponent extends React.Component<IColor3LineComponentProps, { isExpanded: boolean, color: Color3 }> {
+export class Color3LineComponent extends React.Component<IColor3LineComponentProps, { isExpanded: boolean, color: Color3 | Color4, colorText: string }> {
     private _localChange = false;
     constructor(props: IColor3LineComponentProps) {
         super(props);
 
-        this.state = { isExpanded: false, color: this.props.target[this.props.propertyName].clone() };
+        const typeName = typeof (this.props.target[this.props.propertyName]);
+        if (typeName === "string") {
+            let colorConverted = this.convertToColor3(this.props.target[this.props.propertyName]);
+            this.state = { isExpanded: false, color: colorConverted, colorText: this.props.target[this.props.propertyName] };
+        } else {
+            this.state = { isExpanded: false, color: this.props.target[this.props.propertyName].clone(), colorText: this.props.target[this.props.propertyName].toHexString() };
+        }
 
         if (props.isLinear) {
             this.state.color.toGammaSpaceToRef(this.state.color);
         }
 
         props.target._isLinearColor = props.isLinear; // so that replayRecorder can append toLinearSpace() as appropriate
+        this._colorPickerOpen = false;
+    }
+
+    private convertToColor3(color: string) {
+        if (color === "" || color === "transparent" || color === "Transparent") {

```

Comment:
you can use a regex or automatically convert the string to lowercase. `Transparent` is technically invalid.

Reply:
I think I'm not going to even include it since you're right since "Transparent" is not valid.